provider/pkg/provider/provider.go

```diff
@@ -400,10 +400,15 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
 	// Configure client-go using provided or ambient kubeconfig file.
 	//
 
+	if defaultNamespace := vars["kubernetes:config:namespace"]; defaultNamespace != "" {
+		k.defaultNamespace = defaultNamespace
+	}
+
 	// Compute config overrides.
 	overrides := &clientcmd.ConfigOverrides{
 		Context: clientapi.Context{
-			Cluster: vars["kubernetes:config:cluster"],
+			Cluster:   vars["kubernetes:config:cluster"],
+			Namespace: k.defaultNamespace,
```

Comment:
Do we need to handle anything related to "" being "default" etc. here? I believe the context resolves empty to default but wanted to call out..

Reply:
AFAIK, this is fine.